zowe/zowe-client-python-sdk
# Add Secrets SDK with Rust keyring bindings

**What It Does**
Resolves #208 by adding Python to Rust bindings that use the same keyring backend as the Node SDK.
Replaced all references to the `keyring` package with our own `keyring` subpackage in the Secrets SDK.
Added Secrets SDK as an optional dependency of the core SDK that can be installed with the `secrets` feature.

* [x] Merge https://github.com/zowe/zowe-cli/pull/1844
* [x] Update zowe-cli branch in Cargo.toml
* [x] Merge https://github.com/zowe/zowe-client-python-sdk/pull/223
* [ ] Add "release-dev" label to this PR

**How to Test**
Reinstall dependencies after pulling this branch. The install script for Secrets SDK should build the keyring binary.

Manually test the methods of the `keyring` package with this script:
```python
from zowe.core_for_zowe_sdk import CredentialManager
from zowe.secrets_for_zowe_sdk import keyring
# Store short password using keyring module directly
password = "abc123"
keyring.set_password("Test", "ShortPassword", password)
assert keyring.get_password("Test", "ShortPassword") == password
assert keyring.delete_password("Test", "ShortPassword")
assert keyring.get_password("Test", "ShortPassword") is None
# Store long password using CredentialManager wrapper
password *= 512
CredentialManager._set_credential("Test", "LongPassword", password)
assert CredentialManager._get_credential("Test", "LongPassword") == password
CredentialManager._delete_credential("Test", "LongPassword")
assert CredentialManager._get_credential("Test", "LongPassword") is None
```

**Review Checklist**
I certify that I have:


**Additional Comments**
Coauthored with @traeok 🙂
